R/pairwise-comparisons.R

```diff
@@ -0,0 +1,594 @@
+#' \link[surveillance]{permutationTest} from the `surveillance` package
+#'  by Michael Höhle,
+#' Andrea Riebler and Michaela Paul.
+#'
+#' @param scores A data.frame of unsummarised scores as produced by
+#' \code{\link{eval_forecasts}}
+#' @param metric A character vector of length one with the metric to do
+#' the comparison on.
+#' @param by character vector of columns to group scoring by. This should be the
+#' lowest level of grouping possible, i.e. the unit of the individual
+#' observation. This is important as many functions work on individual
+#' observations. If you want a different level of aggregation, you should use
+#' \code{summarise_by} to aggregate the individual scores.
+#' Also not that the pit will be computed using \code{summarise_by}
+#' instead of \code{by}
+#' @param summarise_by character vector of columns to group the summary by. By
+#' default, this is equal to `by` and no summary takes place.
+#' But sometimes you may want to to summarise
+#' over categories different from the scoring.
+#' \code{summarise_by} is also the grouping level used to compute
+#' (and possibly plot) the probability integral transform(pit).
+#' @param test_options list with options to pass down to `compare_two_models()`
+#' @param baseline character vector of length one that deontes
+#' the baseline model against which to compare other models.
+#' @return A ggplot2 object with a coloured table of summarised scores
+#' @importFrom data.table as.data.table data.table setnames copy
+#' @importFrom stats sd rbinom wilcox.test p.adjust
+#' @importFrom utils combn
+#' @export
+#' @author Johannes Bracher, https://jbracher.github.io/
+#' @author Nikos Bosse
+#' @examples
+#' df <- data.frame(model = rep(c("model1", "model2", "model3"), each = 10),
+#'                  date = as.Date("2020-01-01") + rep(1:5, each = 2),
+#'                  location = c(1, 2),
+#'                  interval_score = (abs(rnorm(30))),
+#'                  aem = (abs(rnorm(30))))
+#'
+#' res <- scoringutils::pairwise_comparison(df,
+#'                            baseline = "model1")
+#' scoringutils::plot_pairwise_comparison(res)
+#'
+#' eval <- scoringutils::eval_forecasts(scoringutils::range_example_data_long)
+#' pairwise <- pairwise_comparison(eval, summarise_by = c("model"))
+#'
+
+
+
+
+pairwise_comparison <- function(scores,
+                                metric = "interval_score", # maybe the default can happen automatically,
+                                test_options = list(oneSided = FALSE,
+                                                    test_type = c("non_parametric", "permuation"),
+                                                    n_permutations = 999),
+                                baseline = NULL,
+                                by = NULL,
+                                summarise_by = c("model")) {
+
+  scores <- data.table::as.data.table(scores)
+
+  # update test options
+  test_options <- update_list(defaults = list(oneSided = FALSE,
```

Comment:
As this is used here might not need to specify them in the arg list for the main function and instead document their settings? It feels like a repeat of information.